CommonTools/TriggerUtils/src/GenericTriggerEventFlag.cc

```diff
@@ -203,7 +203,7 @@ void GenericTriggerEventFlag::initRun(const edm::Run& run, const edm::EventSetup
     if (stage2_) {
       l1uGt_->retrieveL1Setup(setup);
 
-      const std::vector<std::pair<std::string, int> > prescales = l1uGt_->prescales();
+      const std::vector<std::pair<std::string, double> > prescales = l1uGt_->prescales();
```

Comment:
did it really need to be a double (not float)?

Reply:
As far as I know the hardware stores non-integer prescales in units of 1/100 .

The L1T group has decreed that using that also in software was too complicated for the average physicist in CMS - and storing integers and dividing by 100 on the fly was too complicated for them, apparently.

Reply:
Thanks @fwyzard and @Martin-Grunewald 
let's discuss what we we do with this.  It depends on how the HLT would/could possibly use these fractional prescales.  Maybe not at all. In any case an overloaded or templated function(s) could be a solution.

Currently the HLT code still compiles simply because the function `HLTPrescaleProvider::prescaleValues ` return type is still left as `std::pair<int, int>`, effectively taking the int value of the l1t prescale.  We don't want to leave it as is.  Or maybe we do ?  if the HLT does not need the l1t fractional prescale, as I think it does not.   But maybe I am wrong.  
Please comment.

Reply:
https://github.com/cms-sw/cmssw/pull/32377#discussion_r535361296

Hi @fwyzard 
I am not aware of any "decree".  What exactly do you have in mind?  Storing int as (l1t_fract_prescale * 100) ?

Reply:
> What exactly do you have in mind? Storing int as (l1t_fract_prescale * 100) ?

Yes, IMHO that would have made more sense than storing doubles, and would have allowed exposing the value with more than one interface; e.g. both as a floating point and as a fixed decimal.

But I assume that the CondFormats is using doubles, so it would make little sense to undo the change now.

Reply:
>  It depends on how the HLT would/could possibly use these fractional prescales. Maybe not at all.

The user is not the HLT, is people analysing the data that need to access both L1T and HLT prescales.

So, the values returned should be the correct ones, *i.e.* the values used during data taking.

Reply:
> But I assume that the CondFormats is using doubles, so it would make little sense to undo the change now.

Correct. The payloads of the L1T Menu which among other things contain prescales which are now fractional and of format `double` are stored in the DB as such, and this is reflected in the CondFormats, yes that is correct.

Reply:
And this is reflected in the external UTM library which is used by O2O, uGT emulator (not to mention the L1T Menu Editor, DQM, etc ).

Reply:
> The user is not the HLT, is people analysing the data that need to access both L1T and HLT prescales.
>
>So, the values returned should be the correct ones, i.e. the values used during data taking.

Sure.  Thinking again about overloading and templating the methods, I think we don't need them.  I would suggest we change `HLTPrescaleProvider::prescaleValues` to return <l1_prescale,HLT_prescale> as  <double,int> and that we accordingly change the functions down-stream.   

Namely, the  history of l1t prescales is fully recreated in the new L1T tag in the DB where l1t prescales are stored as <double> for all the runs taken.

Reply:
> And this is reflected in the external UTM library which is used by O2O, uGT emulator (not to mention the L1T Menu Editor, DQM, etc ).

Exactly, which is what I was writing to @slava77: the L1T group has decided to use `double` instead of a more precise and  compact format "because reasons", so CMS has to adapt.

Reply:
I made a general comment here, but  I think we should consider renaming prescalesValues to something else. As we can fix the (very limited) downstream code that uses it. But we cant fix all user code we dont know about. And thus we should force it not working. The only way is to change the function name/signature.